tiledb/sm/filesystem/vfs.h

```diff
@@ -161,11 +170,17 @@ struct VFSBase {
 /** The S3 filesystem. */
 #ifdef HAVE_S3
 struct S3_within_VFS {
-  S3 s3_;
   template <typename... Args>
   S3_within_VFS(Args&&... args)
       : s3_(std::forward<Args>(args)...) {
   }
+
+  inline S3* s3() {
+    return &s3_;
+  }
```

Comment:
@eric-hughes-tiledb just wanted to point this out for review, I added this to simplify S3 unit tests a bit in my last commit.

I think this could go away once file systems inherit from a common base class. When that is done I could write a test class that stores a pointer to the FilesystemBase and construct what I need for the test based on the input URI. That said if we don't want to add this accessor for now, I could instead construct `sm::S3` as a member of `S3Test` class and compile it out with preprocessor checks for non-s3 builds until base class work is done.